src/MongoDB.Analyzer/Core/ExpressionProcessor.cs

```diff
@@ -132,7 +172,8 @@ public static (SyntaxNode RewrittenLinqExpression, ConstantsMapper ConstantsMapp
             var symbolInfo = rewriteContext.SemanticModel.GetSymbolInfo(nodeToHandle);
             if (symbolInfo.Symbol == null)
             {
-                return default;
+                nodesRemapping.Clear();
```

Comment:
I am not a fan of indicating an error by clearing accumulated results. Let's have this method return a boolean indicating it's success.

Reply:
Done